build/main.go

```diff
@@ -92,6 +99,22 @@ type NativeAmount struct {
 	Amount string
 }
 
+type Path struct {
+	Assets []Asset
+}
+
+// PathDestination is a mutator that configures a path_payment's destination asset and amount
+type PathDestination struct {
```

Comment:
I'm only referring to the `PathDestination` struct in the comment above... It doesn't deal source portion of the payment.

So, I tried to design this package so that the onramp is gradual...  Lessons learned crafting one kind of transaction can be used when crafting other transactions.  IMO, that suggests that rather than replacing `Destination`, we introduce a modifier that augments the built transaction.  Here's the flow I see:

1.  If a payment has not destination information (asset/amount) an error occurs
2.  If a payment only has destination information, a `PaymentOp` is created when applied
3.  If a payment has source information, instead a `PathPaymentOp` will be created when applied, with a default empty path
4.  If a payment has path information, use it instead of the default.

By doing this, my hunch is that developers will be able to gradually learn the api, eventually learning how to craft more advanced transactions by adding new mutators to their repertoire instead of having to learn separate "constellations" of structs that only interact with one another to perform a given advanced transaction.

Doe that make sense?